test/methods/gd_lipschitz_approximation.jl

```diff
@@ -0,0 +1,172 @@
+# OptimizationMethods.jl
+
+module TestGDLipschitzApproximation
+
+using Test, OptimizationMethods, LinearAlgebra, Random
+
+@testset "Method: GD Lipschitz Approximation" begin
+
+    #Testing Context 
+    Random.seed!(1010)
+
+    ##########################################
+    # Test struct properties
+    ##########################################
+
+    ## test if method struct is defined 
+    @test @isdefined LipschitzApproxGD
+    
+    ## test supertype of method struct 
+    @test supertype(LipschitzApproxGD) == 
+        OptimizationMethods.AbstractOptimizerData
+
+    ## Test Field Names and Types 
+    field_info(type::T) where T = [
+        [:name, String], 
+        [:init_stepsize, type], 
+        [:threshold, type], 
```

Comment:
Should incorporate the other field values if you think they are appropriate in this test

Reply:
I do not know how I missed them. Done.